### DISCUSSION TITLE
DNS lookup

### DISCUSSION THREAD

**Discussion by SixoneKui** [hyperium/hyper#3284] [Q&A]
Hi, I am moving from js to rust.
With the JS HTTPs module, there is a way to set up a lookup function in http connection and thereby, with any domain, I can specify the IP ( the usecase is cloudflare , which matches multiple IPs to a domain but I want only connect one ).

Can I do this with hyper?

In addition, how do I enable "keepalive"? Just add the header?
Thank you very much.

**Comment by seanmonstar** [comment]
You probably want to use [reqwest](https://crates.io/crates/reqwest), which does have a [resolve](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.resolve) options. It will use keep-alive automatically.
